File: 3158-find-the-xor-of-numbers-which-appear-twice/3158-find-the-xor-of-numbers-which-appear-twice.java
class Solution {
    public int duplicateNumbersXOR(int[] nums) {
        HashMap<Integer, Integer> freq=new HashMap<>();
        for(int i: nums){
            freq.put(i, freq.getOrDefault(i, 0)+1);
        }
        int ans=0;
        for(Map.Entry<Integer, Integer> entry: freq.entrySet()){
            if(entry.getValue()==2){
                ans^=entry.getKey();
            }
        }
        return ans;
    }
}

File: 3158-find-the-xor-of-numbers-which-appear-twice/3158-find-the-xor-of-numbers-which-appear-twice.cpp
class Solution {
public:
    int duplicateNumbersXOR(vector<int>& nums) {
        map<int, int> freq;
        for(int i: nums){
            freq[i]++;
        }
        int ans=0;
        for(auto j: freq){
            if(j.second==2){
                ans^=j.first;
            }
        }
        return ans;
    }
};